### PR TITLE
Deploy Appwrite on DigitalOcean

### PR DESCRIPTION
**Pull request by spiritbro1**
tutorial to ```Deploy Appwrite on DigitalOcean``` done for issue #546

### CODE REVIEW

File: docs/deployment/digitalocean.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+![image](https://imgur.com/download/OTKn3p8)
+
+to get started sign up with your email address and password
+
+![image](https://imgur.com//download/Wva5tOi)
+
+after that create a project
+
+![image](https://imgur.com/download/6LKlQoP)
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:127]: Maybe edit "image" into a text so people with screen readers can also see what is on the image, also host the images inside the appwrite repository. Make sure that you have a consistent size for all images. I advise you to read through the last two points of Eldad in #546:
* Include only images with specific size (1200x800px) in PNG format
* Include all images in the same directory as your markdown file For example: my-new-appwrite-tutorials/image-1.png

File: docs/deployment/digitalocean.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+after that if you go to this screen you are successfully deploying your appwrite server into digitalocean
+
+![image](https://imgur.com/download/gaoBGGg)
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:131]: Maybe edit "image" into a text so people with screen readers can also see what is on the image, also host the images inside the appwrite repository. Make sure that you have a consistent size for all images. I advise you to read through the last two points of Eldad in #546:
* Include only images with specific size (1200x800px) in PNG format
* Include all images in the same directory as your markdown file For example: my-new-appwrite-tutorials/image-1.png

File: docs/deployment/digitalocean.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    -e version=0.6.2 \
+    appwrite/install
+```
+![image](https://imgur.com/download/l7R5FZf)
+
+input this according to your need for this tutorial we gonna go with the default
+
+![image](https://imgur.com/download/ecCZ1O5)
+
+after its all done exit your terminal and go to your browser and type your droplet ip address it usually take 1 minute or
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:116]: Capitalization and punctuation

File: docs/deployment/digitalocean.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+input this according to your need for this tutorial we gonna go with the default
+
+![image](https://imgur.com/download/ecCZ1O5)
+
+after its all done exit your terminal and go to your browser and type your droplet ip address it usually take 1 minute or 
+less to start your server you will see this in your browser if you successfully run your server
+
+![image](https://imgur.com/download/OTKn3p8)
+
+to get started sign up with your email address and password
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:121]: Capitalization and you are using `:` at some places for images, and some places don't have them. Please be consistent.

File: docs/deployment/digitalocean.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+![image](https://imgur.com//download/Wva5tOi)
+
+after that create a project
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:125]: Capitalization and you are using `:` at some places for images, and some places don't have them. Please be consistent.

File: docs/deployment/digitalocean.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+![image](https://imgur.com/download/6LKlQoP)
+
+after that if you go to this screen you are successfully deploying your appwrite server into digitalocean
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:129]: Capitalization, punctuation and you are using `:` at some places for images, and some places don't have them. Please be consistent.

File: docs/deployment/digitalocean.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+```bash
+docker run -it --rm \
+    --volume /var/run/docker.sock:/var/run/docker.sock \
+    --volume "$(pwd)"/appwrite:/install/appwrite:rw \
+    -e version=0.6.2 \
+    appwrite/install
+```
+![image](https://imgur.com/download/l7R5FZf)
+
+input this according to your need for this tutorial we gonna go with the default
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:112]: Capitalization, punctuation and you are using `:` at some places for images, and some places don't have them. Please be consistent.

File: docs/deployment/digitalocean.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+after that create a project
+
+![image](https://imgur.com/download/6LKlQoP)
+
+after that if you go to this screen you are successfully deploying your appwrite server into digitalocean
+
+![image](https://imgur.com/download/gaoBGGg)
+
+# What's next?
+Congratulations! You've just deploy appwrite into digitalocean.
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:134]: This lines does not sound that good, please rewrite it.

File: docs/deployment/digitalocean.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+Good Luck on your future developments using Appwrite! If you need any help feel free to join the [Discord](https://discord.com/invite/ZFwqr3S) or refer to the [Appwrite Documentation](https://appwrite.io/docs).
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:136]:
```suggestion
Good luck on your future development using Appwrite! If you need any help, feel free to join the [Discord](https://appwrite.io/discord) or refer to the [Appwrite Documentation](https://appwrite.io/docs). 
```

**Review by Kolkies — CHANGES_REQUESTED** [review]

File: docs/deployment/digitalocean.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+## Prequisites
+
+- OS 
+- terminal 
+
+## Getting started
+
+First of all you should get a digitalocean account you can use my referral link [here](https://m.do.co/c/35524e9707b0), to get an account, you will
+automatically get $100 to spend in digitalocean, don't forget to have $5 with you in credit card or paypal to make
```
[the remark this answers is not in the document]
Comment (eldadfux) [reply docs/deployment/digitalocean.md:11]: Yes, not sure this is the correct phrasing. Maybe just have a statement about having some credits in your account to be able to set up a server. Don't think SPAM got anything todo with it.

File: docs/deployment/digitalocean.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
+# How to deploy appwrite into digitalocean
+
+## Prequisites
+
+- OS 
+- terminal 
+
+## Getting started
+
+First of all you should get a digitalocean account you can use my referral link [here](https://m.do.co/c/35524e9707b0), to get an account, you will
```
[the remark this answers is not in the document]
Comment (eldadfux) [reply docs/deployment/digitalocean.md:10]: This is probably less appropriate inside the open-source version.

[the remark this answers is not in the document]
Comment (Kolkies) [reply docs/deployment/digitalocean.md:10]: We had an internal discussion about that referral link, please remove it.

[the remark this answers is not in the document]
Comment (eldadfux) [reply docs/deployment/digitalocean.md:10]: @spiritbro1 once the review process is completed your'e more than welcome to republish the tutorial on any medium with any referral code you wish, just exclude it from the repo.

File: docs/deployment/digitalocean.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
+# How to deploy appwrite into digitalocean
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: ok fix all digitalocean in commit 4e7b38d88db3337b8b40319c52a06667b2c1d3b2

File: docs/deployment/digitalocean.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+## Prequisites
+
+- OS 
+- terminal
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in c311243ddc01a9a70b6711907b35383a9d77ed4a

File: docs/deployment/digitalocean.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+
+## Getting started
+
+First of all you should get a digitalocean account you can use my referral link [here](https://m.do.co/c/35524e9707b0), to get an account, you will
+automatically get $100 to spend in digitalocean, don't forget to have $5 with you in credit card or paypal to make 
+a pre-payment which is a first payment to help prevent spam by digitalocean, if you already got an account skip this step.
+
+## Step 1 - Create a droplet
+
+What is a droplet? so droplet is like a server that we can use to run the appwrite server so in order to do that we should create one
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in 87c9ee03ea03d7847b02c3727c0380b118f86ce1

File: docs/deployment/digitalocean.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: referral link are deleted and shorten the sentence in 87c9ee03ea03d7847b02c3727c0380b118f86ce1

File: docs/deployment/digitalocean.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in 87c9ee03ea03d7847b02c3727c0380b118f86ce1

File: docs/deployment/digitalocean.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (Kolkies) [reply docs/deployment/digitalocean.md]: Only this terminal should start with a capitalized letter, not all of them.

File: docs/deployment/digitalocean.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+follow the step below
+
+After you create an account or login you will get a project in the homepage like this:
+
+![image](https://imgur.com/download/ZQ2FUjX)
+
+my project name is panda you can change the name of your project to whatever you want it doesn't matter
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fix in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+- OS 
+- Terminal 
+
+## Getting started
+
+First of all you should get a DigitalOcean account, if you already got an account let's continue to setup our appwrite server.
+
+## Step 1 - Create a Droplet
+
+What is a Droplet? DigitalOcean Droplet are Linux-based virtual machines (VMs) that run on top of virtualized hardware. Each Droplet you create is a new server you can use, either standalone or as part of a larger, cloud-based infrastructure.
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:14]:
```suggestion
What is a Droplet? DigitalOcean Droplets are Linux-based virtual machines (VMs) that run on top of virtualized hardware. Each Droplet is a server that you can use, either standalone or as part of a larger, cloud-based infrastructure.
```

File: docs/deployment/digitalocean.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+ This droplet then can be used to run our appwrite server
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:15]: Random space at the beginning of the line here

File: docs/deployment/digitalocean.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+
+![image](https://imgur.com/download/ZQ2FUjX)
+
+my project name is panda you can change the name of your project to whatever you want it doesn't matter
+the most important thing right now is you can see in the middle of your screen there is a button that say "Get started with a droplet"
+like this :
+
+![image](https://imgur.com/download/f9DsKAw)
+
+click that button and you will go to a new page like this:
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+The minimum requirements to run Appwrite is as little as 1 CPU core and 2GB of RAM, and an operating system that supports Docker.
+ We will use the default one using ubuntu 20.04 and RAM 2GB
+
+ ![image](https://imgur.com/download/4bqbGYg)
+ 
+ scroll down into password section you can choose either using ssh keys or password we gonna use ssh keys because it's much more secure:
+
+![image](https://imgur.com/download/IpDbMmJ)
+
+click "new ssh key" and then you will get this window:
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

**Review by Kolkies — CHANGES_REQUESTED** [review]

File: docs/deployment/digitalocean.md [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+click that button and you will go to a new page like this:
+
+![image](https://imgur.com/download/XVRkq8C)
+
+The minimum requirements to run Appwrite is as little as 1 CPU core and 2GB of RAM, and an operating system that supports Docker.
+ We will use the default one using ubuntu 20.04 and RAM 2GB
+
+ ![image](https://imgur.com/download/4bqbGYg)
+ 
+ scroll down into password section you can choose either using ssh keys or password we gonna use ssh keys because it's much more secure:
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+
+![image](https://imgur.com/download/IpDbMmJ)
+
+click "new ssh key" and then you will get this window:
+
+![image](https://imgur.com/download/cUNWz5w)
+
+there is a detailed explanation on how to make ssh key on the right hand side of the window follow that instruction and then copy paste
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+your ssh public key into the box
+
+![image](https://imgur.com/download/qbMltoP)
+
+copy paste the ssh key you get from before like this:
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+
+![image](https://imgur.com/download/Mpl0SfK)
+
+give it a name and click add SSH key now you're done set up your ssh keys scroll down until you see button "Create Droplet" like below:
```
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md (hunk 5 above)
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md (hunk 6 above)
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md (hunk 7 above)
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in cd1f4765b2a92be9c979ec8f3b51de2399068d79

File: docs/deployment/digitalocean.md (hunk 17 above)
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in 10396a29017fdce59f2247fc567994df4608c0c4

File: docs/deployment/digitalocean.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in 10396a29017fdce59f2247fc567994df4608c0c4

**Comment by Kolkies** [comment]
@spiritbro1 After you fixed the images, I'm going to read everything again and if needed I will request changes again.

**Comment by spiritbro1** [comment]
> @spiritbro1 After you fixed the images, I'm going to read everything again and if needed I will request changes again.

sure im still fixing the image

File: docs/deployment/digitalocean.md (hunk 8 above)
Comment (spiritbro1) [reply docs/deployment/digitalocean.md]: fixed in 030cff5e747675c7829fb3267f77eba4d4d720cd

**Comment by spiritbro1** [comment]
https://github.com/appwrite/appwrite/pull/622#issuecomment-703112188 fixed in https://github.com/appwrite/appwrite/pull/622/commits/e08f831784b82c0f117cdd9ec956ab1acfa63d5c

File: docs/deployment/digitalocean.md [hunk 24]
```diff
@@ -0,0 +1,134 @@
...
+
+After that if you go to this screen you are successfully deploying your Appwrite server into DigitalOcean :
+
+![finish deploying](digitalocean-tutorial/16_finish.png)
+
+# Conclusion
+
+Now that you have your Appwrite server deployed, you have many options for the type of application you want to build and the technologies you want to use to create a richer experience.
+
+Good luck on your future development using Appwrite! If you need any help, feel free to join the [Discord](https://Appwrite.io/discord) or refer to the [Appwrite Documentation](https://Appwrite.io/docs).
```
Comment (Kolkies) [inline docs/deployment/digitalocean.md:133]:
```suggestion
Good luck on your future development using Appwrite! If you need any help, feel free to join the [Discord](https://appwrite.io/discord) or refer to the [Appwrite Documentation](https://appwrite.io/docs). 
```

**Review by Kolkies — CHANGES_REQUESTED** [review]

**Review by Kolkies — APPROVED** [review]

**Pull request closed without merging**